### DISCUSSION TITLE
Cannot run maturin

### DISCUSSION THREAD

**Discussion by hgrecco** [PyO3/maturin#1982] [Q&A] [answered]
### Bug Description

Just followed the example given [here](https://pyo3.rs/v0.20.3/) and I get:
```
error: could not compile `smallvec` (lib) due to 146 previous errors
💥 maturin failed
  Caused by: Failed to build a native library through cargo
  Caused by: Cargo build finished with "exit status: 101": `env -u CARGO PYO3_ENVIRONMENT_SIGNATURE="cpython-3.10-64bit" PYO3_PYTHON="/Users/<USERNAME>/anaconda3/envs/sci310/bin/python" PYTHON_SYS_EXECUTABLE="/Users/<USERNAME>/anaconda3/envs/sci310/bin/python" "cargo" "rustc" "--features" "pyo3/extension-module" "--target" "aarch64-apple-darwin" "--message-format" "json-render-diagnostics" "--manifest-path" "/Users/<USERNAME>/Documents/code/string_sum/Cargo.toml" "--lib" "--" "-C" "link-arg=-undefined" "-C" "link-arg=dynamic_lookup" "-C" "link-args=-Wl,-install_name,@rpath/string_sum.cpython-310-darwin.so"`
```

### Your maturin version (`maturin --version`)

1.5.0

### Your Python version (`python -V`)

Python 3.10.13

### Your pip version (`pip -V`)

23.3.1

### What bindings you're using

pyo3

### Does `cargo build` work?

- [ ] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [ ] Yes

### Steps to Reproduce

1. Install clean anaconda python 3.10 environment
2. Install maturin as described [here](https://www.maturin.rs/installation)
3. Run the example described [here](https://pyo3.rs/v0.20.3/)

See full output: https://gist.github.com/hgrecco/db39a34137fa3d9d2ecf6720f4bdc7ac

A few important extra issues:
- I am using Anaconda Python distribution
- Apple M2
- I tried installing maturing using conda, pip, and cargo in clean anaconda environments. I get the same problem everytime.

**Comment by messense** [comment]
Your gist shows a successful run, please provide the correct and detailed error.

**Reply by hgrecco** [reply]
Sorry for the inconvenience. I didn't realize that the code was calling `maturin build ` that actually works. It is `maturin develop` which does not. I updated the gist

**Reply by messense** [reply, marked as answer]
Have you tried `rustup target add aarch64-apple-darwin`? The error message indicates that you haven't installed the Rust target.

**Reply by messense** [reply]
BTW, your Anaconda Python seems to be x86_64 not aarch64, please check that you have install the right Python architecture?

**Comment by hgrecco** [comment]
> Have you tried `rustup target add aarch64-apple-darwin`? The error message indicates that you haven't installed the Rust target.

This did the trick. I recently moved to from old mac to an M2 using time machine and forgot to update the rust target. I have to say that it is really surprising that this has only appeared now because I have compiled a dozen of rust projects since without any problem.

**Comment by hgrecco** [comment]
> BTW, your Anaconda Python seems to be x86_64 not aarch64, please check that you have install the right Python architecture?

$ file -b /Users/<USERNAME>/anaconda3/envs/sci310/bin/python
Mach-O 64-bit executable arm64
